### PR TITLE
[nnx] Add Sphinx Docs

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?

Preview: https://flax--3678.org.readthedocs.build/en/3678/experimental/nnx/index.html

* Adds a landing page (ported from Flax).
* Adds NNX Basics guide (proposed primary guide).
* Adds MNIST Tutorial (ported Linen's Quick Start).

### CODE REVIEW

File: docs/experimental/nnx/mnist_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+@struct.dataclass
+class Metrics(metrics.Collection):
+  accuracy: metrics.Accuracy
+  loss: metrics.Average.from_output('loss')
+```
+
+## 5. Create the `TrainState`
+
+In Flax, a common practice is to use a dataclass to encapsulate the training state, including the step number, parameters, and optimizer state. The [`flax.training.train_state.TrainState`](https://flax.readthedocs.io/en/latest/flax.training.html#train-state) class is ideal for basic use cases, simplifying the process by allowing you to pass a single argument to functions like `train_step`.
```
Comment (superbobry) [inline docs/experimental/nnx/mnist_tutorial.md:134]: Interesting, the fusion of Flax and NNX!

Do we have to use `TrainState` in this example? Does it make sense to make it more self-contained and free of abstractions from non-NNX parts of Flax?

File: docs/experimental/nnx/nnx_basics.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  model = static.merge(state)
+  # 3. Call the Module
+  y = model(x)
+  # 4. Use split to propagate State updates
+  state, _ = model.split()
+  return y, state
+
+y, state = forward(static, state, x=jnp.ones((1, 2)))
+# 5. Update the state of the original Module
+model.update(state)
```
Comment (superbobry) [inline docs/experimental/nnx/nnx_basics.md:237]: How is `model.update(state)` different from `static.merge(state)`?

**Review by superbobry — APPROVED** [review]

File: docs/experimental/nnx/nnx_basics.md (hunk 2 above)
Comment (cgarciae) [reply docs/experimental/nnx/nnx_basics.md:237]: `update` mutable the original references with the new state, `merge` creates new references with the given state.

File: docs/experimental/nnx/mnist_tutorial.md (hunk 1 above)
Comment (cgarciae) [reply docs/experimental/nnx/mnist_tutorial.md:134]: The Functional API works 🙂. However, I've been thinking of creating a Module-based `TrainState` that would be even more convenient that the current Pytree API.

**Pull request merged**